lib/train/transports/azure.rb

```diff
@@ -59,15 +61,17 @@ def platform
         force_platform!('azure', @platform_details)
       end
 
-      def azure_client(klass = ::Azure::Resources::Profiles::Latest::Mgmt::Client)
-        if cache_enabled?(:api_call)
+      def azure_client(klass = ::Azure::Resources::Profiles::Latest::Mgmt::Client, use_cache = true, vault_name = nil)
```

Comment:
I don't think we need the `use_cache` here. You can turn off caching easily by calling:
```
connection.disable_cache(:api_call)
connection.azure_client(...)
```